### PR TITLE
Test suite improvements

### PR DESCRIPTION
**Pull request by guillemj**
This includes cleanups and fixes for issues on kernels that do not support io_uring fully,

### CODE REVIEW

File: test/send_recv.c [hunk 1]
```diff
@@ -68,6 +69,10 @@ static int do_recv(struct io_uring *ring, struct iovec *iov)
 	struct io_uring_cqe *cqe;
 
 	io_uring_wait_cqe(ring, &cqe);
+	if (cqe->res == -EINVAL) {
+		fprintf(stdout, "cqe not supported, skipping\n");
```
Comment (axboe) [inline test/send_recv.c]: recv not supported?

File: test/send_recv.c [hunk 2]
```diff
@@ -156,6 +161,11 @@ static int do_send(void)
 	}
 
 	ret = io_uring_wait_cqe(&ring, &cqe);
+	if (cqe->res == -EINVAL) {
+		fprintf(stdout, "cqe not supported, skipping\n");
```
Comment (axboe) [inline test/send_recv.c]: Ditto, but send

File: test/short-read.c [hunk 3]
```diff
@@ -78,6 +78,10 @@ int main(int argc, char *argv[])
 		return 1;
 	}
 
+	if (cqe->res == -EINVAL) {
```
Comment (axboe) [inline test/short-read.c]: This one seems odd, readv has been supported since day 1?

**Comment by axboe** [comment]
Just a few comments on that one patch, otherwise looks good. I'll pull it once you fix up the wording(s) in that one, thanks!

File: test/short-read.c (hunk 3 above)
Comment (guillemj) [reply test/short-read.c]: Right sorry, this one is failing for me with cqe->res == 0, and I think I might have checked for errno == EINVAL or similar and then this mutated into that after the fact and was left over. Removed now.

**Comment by axboe** [comment]
Thanks, pulled!

**Pull request merged**